internal/controller/bindings/forwarder_controller.go

```diff
@@ -196,7 +197,12 @@ func (r *ForwarderReconciler) update(ctx context.Context, epb *bindingsv1alpha1.
 
 		log.Info("Handling connnection")
 
-		ngrokConn, err := tlsDialer.Dial("tcp", *op.Spec.Binding.IngressEndpoint)
+		ingressEndpoint, err := getIngressEndpointWithFallback(op.Status.BindingsIngressEndpoint, log)
```

Comment:
Should we pull this logic up out of the cnxnHandler? I think this parsing can be once outside of handling each connection.

Reply:
Ah yep good catch, I definitely had tunnel vision there. I was a little worried about making sure connections get updated when that field on the k8sop status does but I don't think moving it up out of the connection handler actually changes that.